packages/taco/src/taco.ts

```diff
@@ -132,29 +216,64 @@ export const encryptWithPublicKey = async (
  * @throws {Error} If the active DKG Ritual cannot be retrieved or decryption process throws an error,
  * an error is thrown.
  */
-export const decrypt = async (
+export function decrypt(
   provider: ethers.providers.Provider,
   domain: Domain,
   messageKit: ThresholdMessageKit,
   context?: ConditionContext,
   porterUris?: string[],
-): Promise<Uint8Array> => {
+): Promise<Uint8Array>;
+
+/**
+ * Decrypts an encrypted message.
+ *
+ * @export
+ * @param {PublicClient} viemPublicClient - Viem PublicClient for network operations
+ * @param {Domain} domain - Represents the logical network in which the decryption will be performed.
+ * Must match the `ritualId`.
+ * @param {ThresholdMessageKit} messageKit - The kit containing the message to be decrypted
+ * @param {ConditionContext} context - Optional context data used for decryption time values for the condition(s) within the `messageKit`.
+ * @param {string[]} [porterUris] - Optional URI(s) for the Porter service. If not provided, a value will be obtained
+ * from the Domain
+ *
+ * @returns {Promise<Uint8Array>} Returns Promise that resolves with a decrypted message
+ *
+ * @throws {Error} If the active DKG Ritual cannot be retrieved or decryption process throws an error,
+ * an error is thrown.
+ */
+export function decrypt(
+  viemPublicClient: PublicClient,
+  domain: Domain,
+  messageKit: ThresholdMessageKit,
+  context?: ConditionContext,
+  porterUris?: string[],
+): Promise<Uint8Array>;
+
+export async function decrypt(
+  providerLike: ProviderLike,
+  domain: Domain,
+  messageKit: ThresholdMessageKit,
+  context?: ConditionContext,
+  porterUris?: string[],
+): Promise<Uint8Array> {
   const porterUrisFull: string[] = porterUris
     ? porterUris
     : await getPorterUris(domain);
   const porter = new PorterClient(porterUrisFull);
 
+  const providerAdapter = await toEthersProvider(providerLike);
```

Comment:
I don't think this needs to be async
```suggestion
  const providerAdapter = toEthersProvider(providerLike);
```

Reply:
Thanks for the catch. This was left by a chance after some refactoring that was async.